src/org/openstreetmap/josm/gui/dialogs/properties/PropertiesDialog.java

```diff
@@ -1347,38 +1359,137 @@ public void actionPerformed(ActionEvent e) {
         }
     }
 
-    static SearchSetting createSearchSetting(String key, Collection<? extends IPrimitive> sel, boolean sameType) {
-        String sep = "";
-        StringBuilder s = new StringBuilder();
-        Set<String> consideredTokens = new TreeSet<>();
-        for (IPrimitive p : sel) {
-            String val = p.get(key);
-            if (val == null || (!sameType && consideredTokens.contains(val))) {
-                continue;
+    class FilterAction extends AbstractAction implements PopupMenuListener {
```

Comment:
Stupid question: is there any particular reason why you aren't using `JosmAction`?

Reply:
Mostly for similarity with all the other actions in the context menu which are also simply using `AbstractAction`. We don't really need any features that `JosmAction` provides (no shortcuts, listeners, etc...) so we wouldn't gain a lot from using it. Setting the icon etc... would be a tiny bit simplified.

Reply:
Fair enough.